src/common/diagnostics.py

```diff
@@ -62,6 +63,8 @@ def setup_diagnostics_variables(model, start_time, options, solver_options):
             atol = solver_options.get('atol', None)
             if (rtol is None) or (atol is None):
                 rtol, atol = model.get_tolerances()
+            if isinstance(atol, numbers.Number): # is atol is scalar, convert to list
```

Comment:
Do we want to handle cases where atol is array(1e-4) and array([1e-4] ?

Reply:
I think we do, since we currently allow such `atol` inputs + the error messages for using a size-mismatched atol point to the fact that `atol = [1e-4]` is allowed.

`Failed to set the solver option 'atol' with msg: atol must be of length one or same as the dimension of the problem.`

Fixed; see latest commit.